test/jdk/javax/swing/JButton/bug4490179.java

```diff
@@ -53,10 +56,8 @@ public static void main(String[] args) throws Exception {
                 frame = new JFrame("bug4490179");
                 button = new JButton("Button");
                 frame.getContentPane().add(button);
-                button.addActionListener(e -> {
-                    if ((e.getModifiers() & InputEvent.BUTTON1_MASK)
-                            != InputEvent.BUTTON1_MASK) {
-                        System.out.println("Status: Failed");
+                button.addActionListener(new ActionListener() {
+                    public void actionPerformed(ActionEvent e) {
                         passed = false;
                     }
```

Comment:
The updated code here is correct; Alexander has restored the condition that was present in the original test which was closed-source and manual.

However, it may be not necessary. Instead, the test should fail immediately if `ActionEvent` contains anything but `BUTTON1_DOWN_MASK`. (Well, cleaning up robot state to release mouse buttons is a good thing to do.